<launch>
  <arg name="serial_no_camera1"    			default="043422251176"/> 			<!-- Note: Replace with actual serial number -->
  <arg name="serial_no_camera2"    			default="032522250392"/> 			<!-- Note: Replace with actual serial number -->
  <arg name="serial_no_camera3"    			default=""/> 			<!-- Note: Replace with actual serial number -->
  <arg name="camera1"              			default="camera1"/>		<!-- Note: Replace with camera name -->
  <arg name="camera2"              			default="camera2"/>		<!-- Note: Replace with camera name -->
  <arg name="camera3"              			default="camera3"/>		<!-- Note: Replace with camera name -->
  <arg name="tf_prefix_camera1"         default="$(arg camera1)"/>
  <arg name="tf_prefix_camera2"         default="$(arg camera2)"/>
  <arg name="tf_prefix_camera3"         default="$(arg camera3)"/>
  <arg name="initial_reset"             default="false"/>
  <arg name="reconnect_timeout"         default="6.0"/>
  
  <!-- depth camera -->
  <arg name="fps_depth1"                default="30"/>
  <arg name="fps_depth2"                default="30"/>
  <arg name="fps_depth3"                default="90"/>
  <arg name="depth_width"               default="848"/>
  <arg name="depth_height"              default="480"/>
  <arg name="enable_depth"              default="false"/>

  <!-- color camera -->
  <arg name="fps_color1"                default="5"/>
  <arg name="fps_color2"                default="5"/>
  <arg name="fps_color3"                default="30"/>
  <arg name="enable_color1"             default="true"/>
  <arg name="enable_color2"             default="true"/>
  <arg name="enable_color3"             default="false"/>
  <arg name="color_width"               default="640"/>
  <arg name="color_height"              default="480"/>
  
  <!-- Infrared cameras -->
  <arg name="infra_width"               default="640"/>
  <arg name="infra_height"              default="480"/>
  <arg name="enable_infra"              default="true"/>
  <arg name="enable_infra1"             default="true"/>
  <arg name="enable_infra2"             default="false"/>
  <arg name="infra_rgb"                 default="false"/>
  <arg name="infra_fps"                 default="5"/>
   
  <!-- Motion module -->
  <arg name="gyro_fps"                  default="200"/>
  <arg name="accel_fps"                 default="250"/>
  <arg name="enable_gyro"               default="true"/>
  <arg name="enable_accel"              default="true"/>

  <!-- enable pointcloud -->
  <arg name="enable_pointcloud"         default="false"/>  

  <group ns="$(arg camera1)">
    <include file="$(find realsense2_camera)/launch/includes/nodelet.launch.xml">
      <arg name="serial_no"             value="$(arg serial_no_camera1)"/>
      <arg name="tf_prefix"         		value="$(arg tf_prefix_camera1)"/>
      <arg name="initial_reset"         value="$(arg initial_reset)"/>
      <arg name="reconnect_timeout"     value="$(arg reconnect_timeout)"/>
      <!-- depth -->
      <arg name="depth_fps"             value="$(arg fps_depth1)"/>
      <arg name="depth_width"           value="$(arg depth_width)"/>
      <arg name="depth_height"          value="$(arg depth_height)"/>
      <arg name="enable_depth"          value="$(arg enable_depth)"/>

      <!-- color camera -->
      <arg name="color_fps"             value="$(arg fps_color1)"/>
      <arg name="color_width"           value="$(arg color_width)"/>
      <arg name="color_height"          value="$(arg color_height)"/>
      <arg name="enable_color"          value="$(arg enable_color1)"/>

      <!-- Motion module -->
      <arg name="gyro_fps"               value="$(arg gyro_fps)"/>
      <arg name="accel_fps"              value="$(arg accel_fps)"/>
      <arg name="enable_gyro"            value="$(arg enable_gyro)"/>
      <arg name="enable_accel"           value="$(arg enable_accel)"/>

      <!-- Infrared -->
      <arg name="infra_width"              value="$(arg infra_width)"/>
      <arg name="infra_height"             value="$(arg infra_height)"/>
      <arg name="enable_infra"             value="$(arg enable_infra)"/>
      <arg name="enable_infra1"            value="$(arg enable_infra1)"/>
      <arg name="enable_infra2"            value="$(arg enable_infra2)"/>
      <arg name="infra_rgb"                value="$(arg infra_rgb)"/>
      <arg name="infra_fps"                value="$(arg infra_fps)"/>      

      <arg name="enable_pointcloud"      value="$(arg enable_pointcloud)"/>
    </include>

    <include file="$(find realsense2_camera)/launch/realsense2_calib.launch">
      <arg name="namespace" value="$(arg camera1)"/>
    </include>

  </group>

  <group ns="$(arg camera2)" if="$(eval serial_no_camera2 != '')">
    <include file="$(find realsense2_camera)/launch/includes/nodelet.launch.xml">
      <arg name="serial_no"             value="$(arg serial_no_camera2)"/>
      <arg name="tf_prefix"		          value="$(arg tf_prefix_camera2)"/>
      <arg name="initial_reset"         value="$(arg initial_reset)"/>
      <arg name="reconnect_timeout"     value="$(arg reconnect_timeout)"/>
      <!-- depth -->
      <arg name="depth_fps"             value="$(arg fps_depth2)"/>
      <arg name="depth_width"           value="$(arg depth_width)"/>
      <arg name="depth_height"          value="$(arg depth_height)"/>
      <arg name="enable_depth"          value="$(arg enable_depth)"/>

      <!-- color camera -->
      <arg name="color_fps"             value="$(arg fps_color2)"/>
      <arg name="color_width"           value="$(arg color_width)"/>
      <arg name="color_height"          value="$(arg color_height)"/>
      <arg name="enable_color"          value="$(arg enable_color2)"/>

      <!-- Motion module -->
      <arg name="gyro_fps"               value="$(arg gyro_fps)"/>
      <arg name="accel_fps"              value="$(arg accel_fps)"/>
      <arg name="enable_gyro"            value="$(arg enable_gyro)"/>
      <arg name="enable_accel"           value="$(arg enable_accel)"/>

      <arg name="enable_pointcloud"      value="$(arg enable_pointcloud)"/>
    </include>
  
    <include file="$(find realsense2_camera)/launch/realsense2_calib.launch">
      <arg name="namespace" value="$(arg camera2)"/>
    </include>
  
  </group>

  <group ns="$(arg camera3)" if="$(eval serial_no_camera3 != '')">
    <include file="$(find realsense2_camera)/launch/includes/nodelet.launch.xml">
      <arg name="serial_no"             value="$(arg serial_no_camera3)"/>
      <arg name="tf_prefix"		          value="$(arg tf_prefix_camera3)"/>
      <arg name="initial_reset"         value="$(arg initial_reset)"/>
      <arg name="reconnect_timeout"     value="$(arg reconnect_timeout)"/>
      <arg name="depth_fps"             value="$(arg fps_depth3)"/>
      <arg name="color_fps"             value="$(arg fps_color3)"/>

    </include>
  </group>
</launch>
